In [1]:
import types
import pandas as pd
from botocore.client import Config
import ibm_boto3

def __iter__(self): return 0

# @hidden_cell
# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.
# You might want to remove those credentials before you share the notebook.
client_a40990c82496432595193cfcacbc4f45 = ibm_boto3.client(service_name='s3',
    ibm_api_key_id='_CEwhwKqcDfIqDZgxidDAZfSlX_KZrhaTTAY91ROapgK',
    ibm_auth_endpoint="https://iam.cloud.ibm.com/oidc/token",
    config=Config(signature_version='oauth'),
    endpoint_url='https://s3.eu-geo.objectstorage.service.networklayer.com')

body = client_a40990c82496432595193cfcacbc4f45.get_object(Bucket='mycapstone-donotdelete-pr-8ui09ipldhstbe',Key='publicwi-fi20190903.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

WIFI = pd.read_csv(body)
WIFI.head()

Wifi Hotspot Name                             Address  \
0         Annerley Library Wifi                    450 Ipswich Road   
1         Ashgrove Library Wifi                   87 Amarina Avenue   
2            Banyo Library Wifi               284 St. Vincents Road   
3    Bracken Ridge Library Wifi   Corner Bracken and Barrett Street   
4  Brisbane Square Library Wifi  Brisbane Square, 266 George Street   

          Suburb   Latitude   Longitude Site  
0       ANNERLEY -27.509423  153.033322  LIB  
1       ASHGROVE -27.443946  152.987098  LIB  
2          BANYO -27.373966  153.078323  LIB  
3  BRACKEN RIDGE -27.317973  153.037874  LIB  
4  BRISBANE CITY -27.470912  153.022460  LIB

In [2]:
WIFI.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 6 columns):
Wifi Hotspot Name    344 non-null object
Address              344 non-null object
Suburb               344 non-null object
Latitude             344 non-null float64
Longitude            344 non-null float64
Site                 344 non-null object
dtypes: float64(2), object(4)
memory usage: 16.2+ KB


In [3]:
print(' There are {} type of sites'.format(
        len(WIFI['Site'].unique()),
        WIFI.shape[0]
    )
)

 There are 21 type of sites


In [4]:
print(WIFI['Site'].unique())

['LIB' 'CBD' 'SOU' 'PAD' 'BRU' 'SAN' 'JAM' 'STO' 'CHI' 'CAX' 'ROM' 'PRK'
 'KPT' 'WYN' 'SLU' 'MKA' 'SBH' 'NUN' 'BFT' 'CALI' 'SGHL']


In [5]:
print('In Brisbane, {} Suburban areas have free access to Internet.'.format(
        len(WIFI['Suburb'].unique()),
        WIFI.shape[0]
    )
)

In Brisbane, 57 Suburban areas have free access to Internet.


In [6]:
print(WIFI['Suburb'].unique())

['ANNERLEY' 'ASHGROVE' 'BANYO' 'BRACKEN RIDGE' 'BRISBANE CITY' 'BALMORAL'
 'CARINA' 'CARINDALE' 'CHERMSIDE' 'MOOROOKA' 'COOPERS PLAINS' 'CORINDA'
 'EVERTON PARK' 'FAIRFIELD' 'UPPER MOUNT GRAVATT' 'GRANGE' 'HAMILTON'
 'HOLLAND PARK' 'INALA' 'INDOOROOPILLY' 'KENMORE' 'MITCHELTON'
 'MOUNT COOT-THA' 'MOUNT GRAVATT EAST' 'MOUNT OMMANEY' 'NEW FARM' 'NUNDAH'
 'SANDGATE' 'STONES CORNER' 'SUNNYBANK HILLS' 'TOOWONG' 'WEST END'
 'WYNNUM' 'ZILLMERE' 'FORTITUDE VALLEY' 'PADDINGTON' 'PETRIE TERRACE'
 'GREENSLOPES' 'SPRING HILL' 'SOUTH BRISBANE' 'NEWSTEAD' 'FITZGIBBON'
 'WOOLOOWIN' 'ASCOT' 'ST LUCIA' 'SEVENTEEN MILE ROCKS' 'BELLBOWRIE'
 'FOREST LAKE' 'CALAMVALE' 'MACGREGOR' 'KANGAROO POINT' 'BULIMBA'
 'HAWTHORNE' 'MILTON' 'EAST BRISBANE' 'NORMAN PARK' 'TENERIFFE']


In [7]:

body = client_a40990c82496432595193cfcacbc4f45.get_object(Bucket='mycapstone-donotdelete-pr-8ui09ipldhstbe',Key='brisbane-city-council-libraries-information-september2019.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

LIB = pd.read_csv(body)
LIB.head()


Venue Name   Latitude   Longitude  \
0         Annerley Library -27.509341  153.033310   
1         Ashgrove Library -27.443876  152.987086   
2            Banyo Library -27.374391  153.078259   
3    Bracken Ridge Library -27.317963  153.037889   
4  Brisbane Square Library -27.470824  153.022451   

                                       Venue address  \
0                      450 Ipswich Rd, Annerley 4103   
1                      87 Amarina Ave, Ashgrove 4060   
2                     284 St Vincents Rd, Banyo 4014   
3  Corner Bracken and Barrett Streets, Bracken Ri...   
4               266 George St, Brisbane City, Q 4000   

                                       Opening hours  \
0  Closed on public holidays\r; Monday, Tuesday a...   
1  Closed on public holidays\r; Monday, Wednesday...   
2  Closed on public holidays\r; Monday and Sunday...   
3  Closed on public holidays\r\nMonday and Sunday...   
4  Closed on public holidays\nMonday: 9am - 6pm\n...   

                     Library facilities and services            Phone number  \
0  Books, Magazines and Newspapers; eBooks, Digit...  Council - 07 3403 8888   
1  Books, Magazines and Newspapers; eBooks, Digit...  Council - 07 3403 8888   
2  Books, Magazines and Newspapers; eBooks, Digit...  Council - 07 3403 8888   
3  Books, Magazines and Newspapers\neBooks, Digit...  Council - 07 3403 8888   
4  Books, Magazines and Newspapers; eBooks, Digit...  Council - 07 3403 8888   

                 Parking                           Meeting Room Information  
0  On street, off street                                                NaN  
1  On street, off street  The Ashgrove Library meeting room is available...  
2  On street, off street  The Banyo Library meeting rooms are available ...  
3  On street, off street  The Bracken Ridge Library meeting room is avai...  
4   No dedicated parking  The Brisbane Square Library's 3 meeting rooms ...

In [8]:
LIB.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33 entries, 0 to 32
Data columns (total 9 columns):
Venue Name                         33 non-null object
Latitude                           33 non-null float64
Longitude                          33 non-null float64
Venue address                      33 non-null object
Opening hours                      33 non-null object
Library facilities and services    33 non-null object
Phone number                       33 non-null object
Parking                            33 non-null object
Meeting Room Information           20 non-null object
dtypes: float64(2), object(7)
memory usage: 2.4+ KB


In [9]:
body = client_a40990c82496432595193cfcacbc4f45.get_object(Bucket='mycapstone-donotdelete-pr-8ui09ipldhstbe',Key='Park_%E2%80%94_Locations.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

PARKS = pd.read_csv(body)
PARKS.head()


OBJECTID PARK_NUMBER            PARK_NAME SAP_ASSET_ID HOUSE_NUMBER  \
0         1       D1573    THOMAS PLACE PARK    B-RE-0458         151A   
1         2       D1575   SCARLET PLACE PARK    B-RE-0457           36   
2         3       D1576      PAT RAFTER PARK    B-RE-0456           20   
3         4       D1578   PARKWAY PLACE PARK    B-RE-0455            6   
4         5       D1579  WINROCK STREET PARK    B-RE-0454           80   

       STREET_ADDRESS        SUBURB  POST_CODE        LONG        LAT  \
0       ANCASTER ROAD  UPPER KEDRON       4055  152.915628 -27.410609   
1       SCARLET PLACE   FOREST LAKE       4078  152.971609 -27.611549   
2     MEDALLION PLACE    CARSELDINE       4034  153.009133 -27.341620   
3     EVERGREEN CLOSE       KENMORE       4069  152.930905 -27.521829   
4  MCCONNELL CRESCENT    BROOKFIELD       4069  152.918997 -27.510579   

   SHAPE_Length    SHAPE_Area  
0    811.437573  12271.650428  
1    181.436083    975.136574  
2   1357.932476  20686.559364  
3    297.243998   4274.945062  
4    265.102156   2633.070456

In [10]:
PARKS.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2167 entries, 0 to 2166
Data columns (total 12 columns):
OBJECTID          2167 non-null int64
PARK_NUMBER       2167 non-null object
PARK_NAME         2167 non-null object
SAP_ASSET_ID      2167 non-null object
HOUSE_NUMBER      2119 non-null object
STREET_ADDRESS    2167 non-null object
SUBURB            2167 non-null object
POST_CODE         2167 non-null int64
LONG              2167 non-null float64
LAT               2167 non-null float64
SHAPE_Length      2167 non-null float64
SHAPE_Area        2167 non-null float64
dtypes: float64(4), int64(2), object(6)
memory usage: 203.2+ KB


In [11]:
print(PARKS['SUBURB'].unique())

['UPPER KEDRON' 'FOREST LAKE' 'CARSELDINE' 'KENMORE' 'BROOKFIELD'
 'WAKERLEY' 'OXLEY' 'MOUNT GRAVATT EAST' 'MANLY WEST' 'ACACIA RIDGE'
 'CARINDALE' 'SEVEN HILLS' 'COOPERS PLAINS' 'FERNY GROVE' 'BALD HILLS'
 'DARRA' 'LUTWYCHE' 'THE GAP' 'SUNNYBANK HILLS' 'FAIRFIELD' 'CORINDA'
 'ROCKLEA' 'ANSTEAD' 'DURACK' 'MORNINGSIDE' 'TARRAGINDI' 'CALAMVALE'
 'KEDRON' 'PARKINSON' 'PULLENVALE' 'TINGALPA' 'HEMMANT' 'CHANDLER'
 'BRIDGEMAN DOWNS' 'KELVIN GROVE' 'TAIGUM' 'BULIMBA' 'FORTITUDE VALLEY'
 'INDOOROOPILLY' 'MILTON' 'PETRIE TERRACE' 'GORDON PARK' 'PADDINGTON'
 'STAFFORD' 'KURABY' 'PALLARA' 'MITCHELTON' 'MURARRIE' 'KEPERRA'
 'BRIGHTON' 'NORMAN PARK' 'WILLAWONG' 'ENOGGERA' 'MACKENZIE'
 'BRACKEN RIDGE' 'LOTA' 'SEVENTEEN MILE ROCKS' 'ASHGROVE' 'SANDGATE'
 'NUNDAH' 'BARDON' 'NORTHGATE' 'WYNNUM' 'MANLY' 'SOUTH BRISBANE' 'ASPLEY'
 'BELMONT' 'SUMNER' 'INALA' 'ALDERLEY' 'RIVERHILLS' 'RANSOME'
 'CHAPEL HILL' 'BALMORAL' 'RUNCORN' 'SALISBURY' 'TENERIFFE' 'GREENSLOPES'
 'ROBERTSON' 'HOLLAND PARK WEST' 'WISHART

In [12]:
print('In Brisbane, {} Suburban areas have parks.'.format(
        len(PARKS['SUBURB'].unique()),
        PARKS.shape[0]
    )
)

In Brisbane, 188 Suburban areas have parks.


In [13]:
P=set(WIFI.Suburb).symmetric_difference(PARKS.SUBURB)
P

{'ACACIA RIDGE',
 'ALBION',
 'ALDERLEY',
 'ALGESTER',
 'ANSTEAD',
 'ARCHERFIELD',
 'ASPLEY',
 'AUCHENFLOWER',
 'BALD HILLS',
 'BARDON',
 'BELMONT',
 'BOONDALL',
 'BOWEN HILLS',
 'BRIDGEMAN DOWNS',
 'BRIGHTON',
 'BRISBANE AIRPORT',
 'BROOKFIELD',
 'BULWER',
 'BURBANK',
 'CAMP HILL',
 'CANNON HILL',
 'CARINA HEIGHTS',
 'CARSELDINE',
 'CHANDLER',
 'CHAPEL HILL',
 'CHELMER',
 'CHERMSIDE WEST',
 'CHUWAR',
 'CLAYFIELD',
 'COORPAROO',
 'COWAN COWAN',
 'DARRA',
 'DEAGON',
 'DOOLANDELLA',
 'DREWVALE',
 'DURACK',
 'DUTTON PARK',
 'EAGLE FARM',
 'EIGHT MILE PLAINS',
 'ELLEN GROVE',
 'ENOGGERA',
 'FERNY GROVE',
 'FIG TREE POCKET',
 'GAYTHORNE',
 'GEEBUNG',
 'GORDON PARK',
 'GRACEVILLE',
 'GUMDALE',
 'HEATHWOOD',
 'HEMMANT',
 'HENDRA',
 'HERSTON',
 'HIGHGATE HILL',
 'HOLLAND PARK WEST',
 'JAMBOREE HEIGHTS',
 'JINDALEE',
 'KALINGA',
 'KARANA DOWNS',
 'KARAWATHA',
 'KEDRON',
 'KELVIN GROVE',
 'KENMORE HILLS',
 'KEPERRA',
 'KHOLO',
 'KOORINGAL',
 'KURABY',
 'LARAPINTA',
 'LOTA',
 'LUTWYCHE',
 'LYTTON'

In [14]:
len(P)

133

Nearly 133 Suburban Areas of Brisbane does not have Free internet access.

In [15]:
!pip install geopy
from geopy.geocoders import Nominatim # 

In [16]:
CLIENT_ID = 'QNTV2U1043QRTXZ3BUOXUBJBCDWLSH1TDZ0SQREEZNT0O30B' # your Foursquare ID
CLIENT_SECRET = 'QQMMJ1FWRRKDGO3PY3TQMINKNEF33SOR314HVLWQ3GOMU52X' # your Foursquare Secret
VERSION = '20200722' # Foursquare API version

print('Credentials:')
print('ID: ' + CLIENT_ID)
print('SECRET:' + CLIENT_SECRET)

Credentials:
ID: QNTV2U1043QRTXZ3BUOXUBJBCDWLSH1TDZ0SQREEZNT0O30B
SECRET:QQMMJ1FWRRKDGO3PY3TQMINKNEF33SOR314HVLWQ3GOMU52X


In [17]:
address = 'Brisbane'

geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

-27.4689682 153.0234991


In [18]:
!pip install folium==0.5.0

In [32]:
import folium 
map_brisbane1 = folium.Map(location=[-27.4689682,153.0234991],zoom_start=10)
for lat,lng,addr,sbrb in zip(WIFI['Latitude'],WIFI['Longitude'],PARKS['STREET_ADDRESS'],PARKS['SUBURB']):
    label = '{}, {}'.format(addr,sbrb)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
    [lat,lng],
    radius=5,
    popup=label,
    color='blue',
    fill=True,
    fill_color='#3186cc',
    fill_opacity=0.7,
    parse_html=False).add_to(map_brisbane1)
map_brisbane1

In [33]:
import folium 
map_brisbane = folium.Map(location=[-27.4689682,153.0234991],zoom_start=10)
for lat,lng,addr,sbrb in zip(PARKS['LAT'],PARKS['LONG'],PARKS['STREET_ADDRESS'],PARKS['SUBURB']):
    label = '{}, {}'.format(addr,sbrb)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
    [lat,lng],
    radius=5,
    popup=label,
    color='RED',
    parse_html=False).add_to(map_brisbane)
map_brisbane

In [27]:
import folium 
map_brisbane = folium.Map(location=[-27.4689682,153.0234991],zoom_start=10)
for lat,lng,addr,sbrb in zip(WIFI['Latitude'],WIFI['Longitude'],WIFI['Address'],WIFI['Suburb']):
    label = '{}, {}'.format(addr,sbrb)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
    [lat,lng],
    radius=5,
    popup=label,
    color='RED',
    fill=False,
    parse_html=False).add_to(map_brisbane)
map_brisbane

In [21]:
import requests
import json
from pandas.io.json import json_normalize 

In [22]:
suburb_latitude = PARKS['LAT'][0] # neighborhood latitude value
suburb_longitude = PARKS['LONG'][0] # neighborhood longitude value

suburb_name = PARKS['SUBURB'][0] # neighborhood name

print('Latitude and longitude values of {} are {}, {}.'.format(suburb_name, 
                                                               suburb_latitude, 
                                                               suburb_longitude))

Latitude and longitude values of UPPER KEDRON are -27.41060895, 152.915628053.


In [28]:
LIMIT = 100
radius = 500
url = 'https://api.foursquare.com/v2/venues/explore?client_id={}&client_secret={}&ll={},{}&v={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, suburb_latitude, suburb_longitude, VERSION, radius, LIMIT)
results = requests.get(url).json()

def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']
    
venues = results['response']['groups'][0]['items']
    
nearby_venues = json_normalize(venues) # flatten JSON

# filter columns
filtered_columns = ['venue.name', 'venue.categories', 'venue.location.lat', 'venue.location.lng']
nearby_venues =nearby_venues.loc[:, filtered_columns]

# filter the category for each row
nearby_venues['venue.categories'] = nearby_venues.apply(get_category_type, axis=1)

# clean columns
nearby_venues.columns = [col.split(".")[-1] for col in nearby_venues.columns]

nearby_venues

KeyError: "None of [Index(['venue.name', 'venue.categories', 'venue.location.lat',\n       'venue.location.lng'],\n      dtype='object')] are in the [columns]"

In [ ]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Address',
                             'Suburb', 
                            'Suburb Latitude', 
                            'Suburb Longitude', 
                            'Venue', 
                            'Venue Latitude', 
                            'Venue Longitude', 
                            'Venue Category']
    
    return(nearby_venues)        

In [ ]:
brisbane_venues_df = getNearbyVenues(
                                   names=PARKS['Suburb'],
                                   latitudes=WIFI['Latitude'],
                                   longitudes=WIFI['Longitude']
                                  )

In [ ]:
print(brisbane_venues_df.shape)
brisbane_venues_df.head()

In [ ]:
brisbane_venues_df.groupby('Venue').count()

In [ ]:
print('There are {} uniques categories.'.format(len(brisbane_venues_df['Venue Category'].unique())))

In [ ]:
# one hot encoding
brisbane_onehot = pd.get_dummies(brisbane_venues_df[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
brisbane_onehot['Suburb'] = brisbane_venues_df['Suburb'] 

# move neighborhood column to the first column
fixed_columns = [brisbane_onehot.columns[-1]] + list(brisbane_onehot.columns[:-1])
brisbane_onehot = brisbane_onehot[fixed_columns]

brisbane_onehot.head()

In [ ]:
brisbane_grouped = brisbane_onehot.groupby('Suburb').mean().reset_index()
brisbane_grouped

In [ ]:
num_top_venues = 5

for suburb in brisbane_grouped['Suburb']:
    print("----"+suburb+"----")
    temp = brisbane_grouped[brisbane_grouped['Suburb'] == suburb].T.reset_index()
    temp.columns = ['venue','freq']
    temp = temp.iloc[1:]
    temp['freq'] = temp['freq'].astype(float)
    temp = temp.round({'freq': 2})
    print(temp.sort_values('freq', ascending=False).reset_index(drop=True).head(num_top_venues))
    print('\n')

In [ ]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

In [ ]:
import numpy as np

num_top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Suburb']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
brisbane_venues_sorted = pd.DataFrame(columns=columns)
brisbane_venues_sorted['Suburb'] = brisbane_grouped['Suburb']

for ind in np.arange(brisbane_grouped.shape[0]):
    brisbane_venues_sorted.iloc[ind, 1:] = return_most_common_venues(brisbane_grouped.iloc[ind, :], num_top_venues)

brisbane_venues_sorted

In [ ]:
from sklearn.cluster import KMeans
# set number of clusters
kclusters = int(len(WIFI["Suburb"].unique()) / 4)
brisbane_grouped_clustering = brisbane_grouped.drop('Suburb', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=1).fit(brisbane_grouped_clustering)


In [ ]:
# add clustering labels
brisbane_venues_sorted.insert(0, 'Cluster Labels', kmeans.labels_)

brisbane_merged = WIFI.copy()
brisbane_merged.rename(columns={'Suburb':'Suburb'}, inplace=True)

brisbane_merged = brisbane_merged.join(brisbane_venues_sorted.set_index('Suburb'), on='Suburb')

brisbane_merged.head() # check the last columns!

In [ ]:
import matplotlib.cm as cm
import matplotlib.colors as colors

map_clusters = folium.Map(location=[WIFI["Latitude"][0], WIFI["Longitude"][0]], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(brisbane_merged['Latitude'], brisbane_merged['Longitude'], brisbane_merged['Suburb'], brisbane_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

In [ ]:
map_brisbane = folium.Map(location=[-27.4689682,153.0234991],zoom_start=10)
for lat,lng,addr,sbrb in zip(PARK['LAT'],PARK['LONG'],PARK['STREET_ADDRESS'],PARK['SUBURB']):
    label = '{}, {}'.format(addr,sbrb)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
    [lat,lng],
    radius=5,
    popup=label,
    color='red',
    fill=True,
    fill_color='#cc3139',
    fill_opacity=0.7,
    parse_html=False).add_to(map_brisbane)
map_brisbane